src/axom/inlet/tests/inlet_function.cpp

```diff
@@ -442,6 +442,48 @@ TEST(inlet_function, dimension_dependent_result)
   EXPECT_FLOAT_EQ(result.vec[1], 1);
 }
 
+struct FooWithScalarFunc
+{
+  std::function<double(double)> bar;
+};
+
+template <>
+struct FromInlet<FooWithScalarFunc>
+{
+  FooWithScalarFunc operator()(const axom::inlet::Table& base)
+  {
+    return {base["foo/bar"]};
+  }
+};
+
+TEST(inlet_function, nested_function_in_struct)
+{
+  std::string testString =
+    "quux = { [0] = { foo = { bar = function (x) return x + 1 end } }, "
+    "         [1] = { foo = { bar = function (x) return x + 3 end } } }";
+  DataStore ds;
+  auto inlet = createBasicInlet(&ds, testString);
+
+  auto& quux_schema = inlet.addStructArray("quux");
+  auto& foo_schema = quux_schema.addStruct("foo");
+
+  foo_schema.addFunction("bar",
+                         FunctionTag::Double,
+                         {FunctionTag::Double},
+                         "bar's description");
+
+  auto foos = inlet["quux"].get<std::vector<FooWithScalarFunc>>();
+  EXPECT_EQ(foos.size(), 2);
+
+  auto& first_func = foos[0].bar;
+  EXPECT_TRUE(static_cast<bool>(first_func));
```

Comment:
Is this a test that it's not a `nullptr`?  If so, that might more clearly show the intent.

Reply:
`std::function` doesn't make it easy to get the actual stored function pointer when it's wrapping a lambda, but I can add a comment clarifying what the check is doing.